src/main/java/org/jenkinsci/plugins/workflow/support/pickles/serialization/PickleResolver.java

```diff
@@ -24,23 +24,38 @@
 
 package org.jenkinsci.plugins.workflow.support.pickles.serialization;
 
-import org.jenkinsci.plugins.workflow.pickles.Pickle;
-import org.jenkinsci.plugins.workflow.support.concurrent.Futures;
 import com.google.common.base.Function;
+import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
-import org.jboss.marshalling.ObjectResolver;
-
+import com.google.common.util.concurrent.MoreExecutors;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
+import jenkins.util.SystemProperties;
+import jenkins.util.Timer;
+import org.jboss.marshalling.ObjectResolver;
 import org.jenkinsci.plugins.workflow.flow.FlowExecutionOwner;
+import org.jenkinsci.plugins.workflow.pickles.Pickle;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
  * {@link ObjectResolver} that resolves {@link DryCapsule} to unpickled objects.
  *
  * @author Kohsuke Kawaguchi
  */
 public class PickleResolver implements ObjectResolver {
+
+    /**
+     * Pickle resolution will fail automatically after this many seconds.
+     * <p>This is intended to prevent Pipeline builds from hanging forever in unusual cases.
+     */
+    @Restricted(NoExternalUse.class)
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "Non-final for modification from script console")
+    public static long RESOLUTION_TIMEOUT_SECONDS = SystemProperties.getLong(PickleResolver.class + ".RESOLUTION_TIMEOUT_SECONDS", TimeUnit.HOURS.toSeconds(1));
```

Comment:
Not sure what value is best here. One hour resolving pickles seemed roughly ok to me. I think we don't really expect `TryRepeatedly` to be used normally anyway now that `ExecutorStepDynamicContext` exists, but there are a couple of [other usages](https://github.com/search?type=code&q=+owner%3Ajenkinsci+tryrepeatedly).